Commands/Selection.lua

```diff
@@ -69,6 +69,64 @@ end
 
 
 
+function HandleDistrCommand(a_Split, a_Player)
+	-- //distr
+	
+	-- TODO: -d option that separates data values.
+	
+	-- Check the selection:
+	local State = GetPlayerState(a_Player)
+	if not(State.Selection:IsValid()) then
+		a_Player:SendMessage(cChatColor.Rose .. "No selection set")
+		return true
+	end
+	
+	-- Get selection information.
+	local World = a_Player:GetWorld()
+	local Area = cBlockArea()
+	Area:Read(World, State.Selection:GetSortedCuboid())
+	local SizeX, SizeY, SizeZ = Area:GetCoordRange()
+	
+	-- Count the blocks.
+	local TotalCount = Area:GetVolume()
+	local BlockCounts = {}
+	
+	for X = 0, SizeX do
+		for Y = 0, SizeY do
+			for Z = 0, SizeZ do
+				local BlockType = Area:GetRelBlockType(X, Y, Z)
+				if BlockCounts[BlockType] == nil
+					then BlockCounts[BlockType] = 1
+					else BlockCounts[BlockType] = BlockCounts[BlockType] + 1
+				end
+			end
+		end
+	end
+	
+	-- Generate the output.
+	-- Sort records by count.
+	SortedBlockCounts = {}
+	for BlockType, BlockCount in pairs(BlockCounts) do
+		table.insert(SortedBlockCounts, {Type=BlockType, Count=BlockCount})
+	end
```

Comment:
This can be speed up. table.insert first has to check how large the table is. If you keep track of how many indexes there are it should be faster:
```lua
local SortedBlockCount, Index = {}, 1
for BlockType, BlockCount in pairs(BlockCounts) do
	SortedBlockCount[Index] = {Type = BlockType, Count = BlockCount}
	Index = Index + 1
end
```